js/data/bucket/circle_bucket.js

```diff
@@ -40,19 +40,9 @@ CircleBucket.prototype.programInterfaces = {
             components: 4,
             type: 'Uint8',
             isLayerConstant: false,
-            value: (
-                'this._premultiplyColor(' +
-                    'layer.getPaintValue("circle-color", globalProperties, featureProperties),' +
-                    'layer.getPaintValue("circle-opacity", globalProperties, featureProperties)' +
-                ')'
-            ),
+            value: 'this._premultiplyColor(layer.getPaintValue("circle-color", globalProperties, featureProperties), 1)',
```

Comment:
Can we do that multiplication within the shader now? It could be a small perf win and a small code complexity win.